res/controllers/midi-components-0.1.js

```
@@ -0,0 +1,1136 @@
+        }
+
+        outGetParameter() {
+            return engine.getParameter(this.group, this.outKey);
+        }
+        outSetParameter(value) {
+            engine.setParameter(this.group, this.outKey, value);
+        }
+        outGetValue() {
+            return engine.getValue(this.group, this.outKey);
+        }
+        outSetValue(value) {
+            engine.setValue(this.group, this.outKey, value);
+        }
+        outToggle() {
+            this.outSetValue(!this.outGetValue());
+        }
+
+        connect() {
+            /**
+            Override this method with a custom one to connect multiple Mixxx COs for a single Component.
+            Add the connection objects to the this.connections array so they all get disconnected just
+            by calling this.disconnect(). This can be helpful for multicolor LEDs that show a
+            different color depending on the state of different Mixxx COs. Refer to
+            SamplerButton.connect() and SamplerButton.output() for an example.
+            **/
+            if (undefined !== this.group &&
+                undefined !== this.outKey &&
+                undefined !== this.output &&
+                typeof this.output === "function") {
+                this.connections[0] = engine.makeConnection(this.group, this.outKey, this.output.bind(this));
+            }
+        }
+        disconnect() {
+            if (this.connections[0] !== undefined) {
+                this.connections.forEach(function(conn) {
+                    conn.disconnect();
+                });
+            }
+        }
+        trigger() {
+            if (this.connections[0] !== undefined) {
+                this.connections.forEach(function(conn) {
+                    conn.trigger();
+                });
+            }
+        }
+        send(value) {
+            if (this.midi === undefined || this.midi[0] === undefined || this.midi[1] === undefined) {
+                return;
+            }
```

Is this expected or an error? If the latter we should throw an exception.